addons/account_edi_ubl_cii/tests/test_ubl_cii.py

```diff
@@ -0,0 +1,70 @@
+from odoo.modules.module import get_module_resource
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+from odoo.tests import tagged
+import base64
+
+
+@tagged('post_install', '-at_install')
+class TestAccountEdiUblCii(AccountTestInvoicingCommon):
+
+    @classmethod
+    def setUpClass(cls, chart_template_ref=None):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+
+        cls.uom_units = cls.env.ref('uom.product_uom_unit')
+        cls.uom_dozens = cls.env.ref('uom.product_uom_dozen')
+
+        cls.displace_prdct = cls.env['product.product'].create({
+            'name': 'Displacement',
+            'uom_id': cls.uom_units.id,
+            'standard_price': 90.0,
+            'property_account_income_id': cls.company_data['default_account_revenue'].id,
+            'property_account_expense_id': cls.company_data['default_account_expense'].id,
+            'taxes_id': [(6, 0, cls.company_data['default_tax_sale'].ids)],
+        })
+
+        cls.place_prdct = cls.env['product.product'].create({
+            'name': 'Placement',
+            'uom_id': cls.uom_units.id,
+            'standard_price': 80.0,
+            'property_account_income_id': cls.company_data['default_account_revenue'].id,
+            'property_account_expense_id': cls.company_data['default_account_expense'].id,
+            'taxes_id': [(6, 0, cls.company_data['default_tax_sale'].ids)],
+        })
+
+    ####################################################
+    # Helpers
+    ####################################################
+
+    def create_invoice_from_file(self, module_name, subfolder, filename):
+        file_path = get_module_resource(module_name, subfolder, filename)
+        file = file_open(file_path, mode='rb').read()
+
+        attachment = self.env['ir.attachment'].create({
+            'name': filename,
+            'datas': base64.encodebytes(file),
+            'res_model': 'account.move',
+        })
+        journal_id = self.company_data['default_journal_sale']
+        action_vals = journal_id.with_context(default_move_type='in_invoice').create_invoice_from_attachment(attachment.ids)
+        return self.env['account.move'].browse(action_vals['res_id'])
+
+    ####################################################
+    # Tests
+    ####################################################
+
+    def test_import_product(self):
+        invoice = self.create_invoice_from_file('account_edi_ubl_cii', 'test_files', 'test_ubl_cii.xml')
+        self.assertRecordValues(
+            invoice.line_ids[2:],
+            [
+                {'product_id': self.place_prdct.id, 'product_uom_id': self.uom_units.id},
+                {'product_id': self.displace_prdct.id, 'product_uom_id': self.uom_units.id},
+                {'product_id': self.displace_prdct.id, 'product_uom_id': self.uom_units.id},
+                {'product_id': self.displace_prdct.id, 'product_uom_id': self.uom_dozens.id}
+            ]
+        )
```

Comment:
Considering the methods this commit is fixing (i.e. `_import_fill_invoice_line_taxes` and `_retrieve_product`). I think the XML file is needed to test the import routine. Maybe I'm missing something, but I feel like the `create` method wouldn't use the previously mentioned methods.
----
The 1st product line is testing that the 'Placement' product is used correctly (instead of the 'Displacement' one).
The 2nd line is testing the case where a product  on the invoice has a UoM with a different category than the one in the DB (here on the XML the product is in 'Hour' units, whereas on the DB the 'Displacement' product is in 'Unit' units). 
The 3rd one is a "sanity check" (i.e. the 2nd one but this time the UoM is the exact same as the product on the DB) 
The 4th one tests the case where the UoM of a product on the invoice is different but in the same category as its corresponding product on the DB (i.e. the 'Dozen' units).
NB: I'll add a comment to the test to make this clear.

Reply:
Sorry, I went way to fast... 
 I meant:
 ```suggestion
    def test_import_product(self):
        line_vals = [
            {'product_id': self.place_prdct.id, 'product_uom_id': self.uom_units.id},
            {'product_id': self.displace_prdct.id, 'product_uom_id': self.uom_units.id},
            {'product_id': self.displace_prdct.id, 'product_uom_id': self.uom_dozens.id}
        ]
        invoice = self.env['account.move'].create({
            'partner_id': self.company_data_2['company'].partner_id,
            'move_type': 'out_invoice',
            'invoice_line_ids': line_vals,
        })
        invoice.action_post()
        facturx_attachment = invoice.edi_document_ids.attachment_id
        xml_tree = etree.fromstring(facturx_attachment.raw)
        new_invoice = self.env.ref('account_edi_facturx.edi_facturx_1_0_05')._create_invoice_from_xml_tree(
            facturx_attachment.name,
            xml_tree,
            invoice.journal_id,
        )
        self.assertRecordValues(new_invoice.invoice_line_ids, line_vals)
```